lib/capstone_project.ex

```diff
@@ -3,16 +3,83 @@ defmodule CapstoneProject do
   Documentation for `CapstoneProject`.
   """
 
+  @type rational :: {integer, integer}
+
+  @doc """
+  Add two rational numbers
+  """
+  @spec add(a :: rational, b :: rational) :: rational
+  def add(a, b) do
+    {{a1, b1}, {a2, b2}} = {a, b}
+    {(a1 * b2 + a2 * b1), b1 * b2}
+    |> reduce()
+  end
+
+  @doc """
+  Subtract two rational numbers
+  """
+  @spec subtract(a :: rational, b :: rational) :: rational
+  def subtract(a, b) do
+    {{a1, b1}, {a2, b2}} = {a, b}
+    {(a1 * b2 - a2 * b1), b1 * b2}
+    |> reduce()
+  end
+
+  @doc """
+  Multiply two rational numbers
+  """
+  @spec multiply(a :: rational, b :: rational) :: rational
+  def multiply(a, b) do
+    {{a1, b1}, {a2, b2}} = {a, b}
+    {a1 * a2, b1 * b2}
+    |> reduce()
+  end
+
   @doc """
-  Hello world.
+  Divide two rational numbers
+  """
+  @spec divide_by(num :: rational, den :: rational) :: rational
+  def divide_by(num, den) do
+    {{a1, b1}, {a2, b2}} = {num, den}
+    {a1 * b2, a2 * b1}
+    |> reduce()
+  end
 
-  ## Examples
+  @doc """
+  Absolute value of a rational number
+  """
+  @spec abs(a :: rational) :: rational
+  def abs(a) do
+    {n, d} = a
+    {Kernel.abs(n), Kernel.abs(d)}
```

Comment:
```suggestion
    [n, d] |> Enum.map(&Kernel.abs/1) |> List.to_tuple()
```

Не уверен, впрочем, что читабельность не снизилась.